cli/azd/cmd/util.go

```diff
@@ -283,7 +288,16 @@ func ensureEnvironmentInitialized(
 	}
 
 	if !hasPrincipalID {
-		principalID, err := azureutil.GetCurrentPrincipalId(ctx, userProfileService)
+		subscriptionId, found := env.Values[environment.SubscriptionIdEnvVarName]
+		if !found {
+			return fmt.Errorf("tried to get principal id without a subscription id selected")
+		}
+		tenantId, err := subResolver.LookupTenant(ctx, subscriptionId)
+		if err != nil {
+			return fmt.Errorf("getting tenant id for subscription %s. Error: %w", subscriptionId, err)
```

Comment:
This may be too verbose. I'd recommend just return `err`.

Based on other usage of `LookupTenant`, we'll already get enough contextual information for the `error`

We already emit context  error like: `resolving user access to subscription '%s' : %w"`

Reply:
So if I see this error `resolving user access to subscription '%s' : %w"` , who do we know who called `LookupTenant` ?